### ISSUE TITLE
Debug mode enabled - product name in pack when stock low

### ISSUE THREAD

**Issue by M-t-t-h** [PrestaShop/PrestaShop#20405]
in first, excuse my english, not my motherlanguare.
Second, it's my first bug report, hope doing it well.

#### Describe the bug
A php error is raising (red warning, no crash) at controllers/admin/AdminOrdersController.php (+/- line 1861) :
`$this->displayWarning($this->trans('This product, included in package (' . $product['product_name'] . ') is out of stock: ', array(), 'Admin.Orderscustomers.Notification') . ' ' . $pack_item['product_name']);` 

#### Expected behavior
have to show the warning about stock

#### Steps to Reproduce

It's happend when we work with pack, when _$display_out_of_stock_warnin = true and $product['current_stock'] <= 0_ 

1. make a order of a pack with an items stock <=0 and display warning
2. go to admin>orders
3. select the order

#### Additional information
PS V: 1.7.6.7

to correct it, change : 
**$pack_item['product_name']** by **$pack_item['name']**

**Comment by florine2623** [comment]
Hi @M-t-t-h ,
I can't reproduce your issue in PS 1.7.6.7.
1. Add one pack of 2 products in cart (1 product has 0 stock)
2. Place the order
3. BO > Orders > Details : No error

In Shop Parameters > Product Settings > Default pack stock management :
![Capture d’écran 2020-07-31 à 13 58 09](https://user-images.githubusercontent.com/16019289/89032949-042d6500-d336-11ea-8eb0-b6ccc387a1e4.png)

Can you screenshot the error ?

Thanks!

**Comment by M-t-t-h** [comment]
Hi,

Please find my screen-shot :
![ps-bugreport](https://user-images.githubusercontent.com/8056358/89034387-21affe00-d339-11ea-83c6-f4f8247f005a.jpg)

My Product who is a pack is set like: 
Qty : -2
Decrement pack only (too)
Pack's item are Qty 0 for almost all 

When i patch the php i have:
![ps-bugreport2](https://user-images.githubusercontent.com/8056358/89034763-e9f58600-d339-11ea-932c-134e62d80ff4.jpg)

It s probably not influancing the issue but it's a multi-store shop

**Comment by florine2623** [comment]
I enabled debut mode and still an't face the error.
Is there anything special about the product in the pack that I should know ?
Quantity of the products/ quantity of the pack/ parameters of the product settings etc

**Comment by M-t-t-h** [comment]
This pack is not special, just set as said...
if you (temporary) add an echo on AdminOrdersController like
```
 foreach ($products as &$product) {
            $pack_items = $product['cache_is_pack'] ? Pack::getItemTable($product['id_product'], $this->context->language->id, true) : array();
            foreach ($pack_items as &$pack_item) {
                $pack_item['current_stock'] = StockAvailable::getQuantityAvailableByProduct($pack_item['id_product'], $pack_item['id_product_attribute'], $pack_item['id_shop']);

echo 'TEST1<br/>';

                // if the current stock requires a warning
                if ($product['current_stock'] <= 0 && $display_out_of_stock_warning) {

echo 'TEST2<br/>';

                    // $this->displayWarning($this->trans('This product, included in package (' . $product['product_name'] . ') is out of stock: ', array(), 'Admin.Orderscustomers.Notification') . ' ' . $pack_item['product_name']);
                    $this->displayWarning($this->trans('This product, included in package (' . $product['product_name'] . ') is out of stock: ', array(), 'Admin.Orderscustomers.Notification') . ' ' . $pack_item['name']);
                }
```
do you have have a return of 'TEST' ? (goal is to arrive at those lines).

In this part, when i `var_dump($pack_item)` i haven't a key named "product_name"

**Comment by florine2623** [comment]
Did you do a migration to get to PS 1.7.6.7 ?

My test shop is a fresh install.
The variable is already `$pack_item['name']`, when I var_dump() I have : `["name"]=> string(27) "Coussin renard Color-White"`

**Comment by M-t-t-h** [comment]
Yes i do !
I used "PrestaShop 1-Click Upgrade module" for it.
(for a tiny upgrade, 1.7.6.5 or 1.7.6.6 to 1.7.6.7)

**Comment by florine2623** [comment]
Ok, you can find the error on the log file of the autoupgrade module.
And you can change the name of the variable. You can find in the attachment the file from a fresh install os PS 1.7.6.7.
[AdminOrdersController.php.zip](https://github.com/PrestaShop/PrestaShop/files/5007412/AdminOrdersController.php.zip)

**Comment by M-t-t-h** [comment]
Ok. Tks for your fast replies and answers.
Sorry to point a knowed issue.

Best regards

**Comment by M-t-t-h** [comment]
Hi, 

This morning, i put your file into my FTP and error came back.
In your file, the line  1861 contains '`.... $pack_item['product_name']);`' again.

Best regards,

**Comment by khouloudbelguith** [comment]
Hi @M-t-t-h,

Thanks for your report.
I did not manage to reproduce the issue after an upgrade from PS1.7.6.5 to PS1.7.6.7 using the 1-click upgrade v4.10.1.
I attached a screen record
https://drive.google.com/file/d/1g65wXqnTYK81tx7er5fcyHZ7o6sx-k5m/view?usp=sharing
What is the exact version of the 1-click upgrade did you use? have you any errors during the upgrade?

Thanks!

**Comment by M-t-t-h** [comment]
Hi,

It's an 4.10.1 version (for upgrade).
It's no more problematic for me cause i change de key on my file.

Just a signal if the "good version" have the bad key...

**Comment by khouloudbelguith** [comment]
@M-t-t-h, as you can see [here](https://github.com/PrestaShop/PrestaShop/blob/1.7.6.7/controllers/admin/AdminOrdersController.php#L1861), it is `$pack_item['product_name'] ` but the issue is not reproduced.

What is the exact version of PHP did you use?

Thanks!

**Comment by M-t-t-h** [comment]
Hi.

If you do a ungly "echo" or "var_dump" at this place, do you realy pass there ?
For my php, it's 7.1 (by ovh.config)

**Comment by khouloudbelguith** [comment]
@M-t-t-h, thanks for these clarifications.
In fact, I manage to reproduce the issue with PS1.7.6.7 fresh install and after an upgrade from PS1.7.6.5.
This issue occurs only when **debug mode is enabled**
![image](https://user-images.githubusercontent.com/16067358/89281385-4e6e5900-d64a-11ea-916c-cf0ae612d3f5.png)


I tried with 177x branch, the order Page is migrated to Symfony, it is OK => Issue fixed
I attached a screen record
https://drive.google.com/file/d/143_eav63HjbeQq-SNIHTcs1LApKC0JL8/view?usp=sharing

Since it is fixed, I will close the issue, feel free to open a new one when needed.

Thanks!
